packages/public/server/src/module/Alias/src/Controller/AliasController.php

```diff
@@ -22,71 +22,149 @@
  */
 namespace Alias\Controller;
 
-use Alias\Exception\AliasNotFoundException;
-use Alias\Exception\CanonicalUrlNotFoundException;
 use Alias;
+use Alias\AliasManagerInterface;
+use Alias\Controller\Plugin\Url;
+use Instance\Manager\InstanceManagerInterface;
+use Normalizer\NormalizerInterface;
+use Uuid\Exception\NotFoundException;
+use Uuid\Manager\UuidManagerInterface;
 use Zend\Http\Request;
 use Zend\Mvc\Controller\AbstractActionController;
-use Zend\Stdlib\ArrayUtils;
+use Zend\Mvc\Router\RouteInterface;
+use Zend\View\Model\JsonModel;
+use Zend\View\Model\ViewModel;
 
 class AliasController extends AbstractActionController
 {
-    use Alias\AliasManagerAwareTrait,
-        \Instance\Manager\InstanceManagerAwareTrait;
+    /** @var AliasManagerInterface */
+    private $aliasManager;
+    /** @var InstanceManagerInterface */
+    private $instanceManager;
+    /** @var UuidManagerInterface */
+    private $uuidManager;
+    /** @var NormalizerInterface */
+    private $normalizer;
+    /** @var RouteInterface */
+    private $router;
 
-    /**
-     * @var mixed
-     */
-    protected $router;
+    public function __construct(
+        AliasManagerInterface $aliasManager,
+        InstanceManagerInterface $instanceManager,
+        UuidManagerInterface $uuidManager,
+        NormalizerInterface $normalizer,
+        RouteInterface $router
+    ) {
+        $this->aliasManager = $aliasManager;
+        $this->instanceManager = $instanceManager;
+        $this->uuidManager = $uuidManager;
+        $this->normalizer = $normalizer;
+        $this->router = $router;
+    }
 
-    public function forwardAction()
+    public function resolveAction()
     {
         $alias = $this->params('alias');
-        $instance = $this->getInstanceManager()->getInstanceFromRequest();
 
-        try {
-            $location = $this->aliasManager->findCanonicalAlias(
-                $alias,
-                $instance
-            );
-            $this->redirect()->toUrl($location);
-            $this->getResponse()->setStatusCode(301);
-            return false;
-        } catch (CanonicalUrlNotFoundException $e) {
+        if (preg_match('/^(?<id>\d+)$/', $alias, $matches)) {
+            return $this->resolveUuid($matches['id']);
+        } else {
+            $url = $this->resolveAlias($alias);
+            return $url === null
+                ? $this->notFoundResponse()
+                : $this->routerResponse($url);
         }
+    }
 
+    private function resolveAlias($alias)
+    {
+        if (preg_match('/(?<id>\d+)\//', $alias, $matches)) {
```

Comment:
This is a problem, when there is a number in the subject's name (e.g "Analysis 1"). I have renamed "Physik" to "Physik 1" -> now https://de.serlo-staging.dev/physik/ resolves to aeneas profile...

Reply:
A first workaround:

```suggestion
        if (preg_match('/\/(?<id>\d+)\//', "/" . $alias, $matches)) {
```

However we still have an issue when the subject's name consists only of digits

Reply:
Fix in https://github.com/serlo/serlo.org/pull/543/commits/8dc26d029a7ebc257abf62fd168ca46d40f33d16